src/build/__init__.py

```diff
@@ -360,6 +382,32 @@ def _handle_backend(self, hook):  # type: (str) -> Iterator[None]
             except Exception as exception:
                 raise BuildBackendException(exception, exc_info=sys.exc_info())
 
+    if sys.version_info[0] >= 3:
+
+        def metadata_path(self, output_directory):  # type: (str) -> Union[pathlib.Path, zipfile.Path]
+            """
+            Returns a :py:class:`pathlib.Path`-like object in the metadata directory.
+
+            :param output_directory: Directory to put the metadata distribution in
+
+
+            .. caution::
+
+               Only available on Python 3.
+            """
+            assert zipfile, 'metadata_path is not available due to missing zipp module'
+            try:
+                return pathlib.Path(typing.cast(str, self.prepare('wheel', output_directory)))
+            except TypeError:
```

Comment:
Catching a type error from the `Path`'s constructor is a roundabout way to check if the return value of `prepare` is not null - I would restructure this as:

```suggestion
            metadata = self.prepare('wheel', output_directory)
            if metadata is not None:
                return pathlib.Path(metadata)
            else:
```

No cast necessary.